File: src/main/java/frc/robot/commands/playCommand.java
package frc.robot.commands;

import java.util.function.Supplier;

import edu.wpi.first.wpilibj2.command.Command;

/**
 * A command that runs another command supplied by a {@link Supplier}. This is
 * useful for reinitalizing a command constructed before calling
 */
public class playCommand extends Command {
    private Supplier<Command> cmd;

    public playCommand(Supplier<Command> cmd) {
        this.cmd = cmd;

    }

    @Override
    public void initialize() {

        addRequirements(cmd.get().getRequirements());
        cmd.get().initialize();
    }

    @Override
    public void execute() {
        cmd.get().execute();
    }

    @Override
    public void end(boolean interrupted) {
        cmd.get().end(interrupted);
    }

    @Override
    public boolean isFinished() {
        return cmd.get().isFinished();
    }
}

File: src/main/kotlin/frc/robot/commands/PlayCommand.kt
package frc.robot.commands

import edu.wpi.first.wpilibj2.command.Command
import java.util.function.Supplier

/**
 * A command that runs another command supplied by a [Supplier]. This is
 * useful for reinitializing a command constructed before calling
 */
class PlayCommand(
    private val cmd: Supplier<Command>
) : Command() {

    override fun initialize() {
        addRequirements(cmd.get().requirements)
        cmd.get().initialize()
    }

    override fun execute() {
        cmd.get().execute()
    }

    override fun end(interrupted: Boolean) {
        cmd.get().end(interrupted)
    }

    override fun isFinished(): Boolean {
        return cmd.get().isFinished
    }
}
